pandas/_libs/tslibs/conversion.pyx

```diff
@@ -167,6 +167,7 @@ def datetime_to_datetime64(values: object[:]):
         int64_t[:] iresult
         npy_datetimestruct dts
         _TSObject _ts
+        bint found_naive = False
```

Comment:
@jbrockmendel is there a reason we cannot remove this routine entirely in favor of using ``array_to_datetime``? can you create an issue about this (though maybe you already did)

Reply:
#24006 does this, but is on hold ATM